docs/runtime-upgrades-and-versioning.md

```diff
@@ -0,0 +1,54 @@
+# Overview 
+
+This diagram shows our compile time and runtime dependencies to allow to guide upgrade process
+
+```plantuml
```

Comment:
You might have to generate an image of this an include as github doesn't support rendering

Reply:
I will add rendered into nix into codespace in next nix related pr (soon). So remote VS code will render this one well.

Reply:
added plugins

Reply:
plantumpl added to nix setup. can open remote devcontainer and observe diagrams. or nix develop .#developer locally